src/Tribe/Views/V2/Views/Recent_Past_View.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * The Past Recent View.
+ *
+ * @package Tribe\Events\Views\V2\Views
+ * @since TBD
+ */
+
+namespace Tribe\Events\Views\V2\Views;
+
+use Tribe\Events\Views\V2\View;
+use Tribe\Events\Views\V2\Views\Traits\List_Behavior;
+use Tribe__Context;
+use Tribe__Events__Main as TEC;
+use Tribe__Events__Rewrite as TEC_Rewrite;
+use Tribe__Utils__Array as Arr;
+use Tribe\Events\Views\V2\Utils;
+
+class Recent_Past_View extends View {
+	use List_Behavior;
+	/**
+	 * Slug for this view
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $slug = 'recent-past';
+
+	/**
+	 * Visibility for this view.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	protected static $publicly_visible = true;
+
+	/**
+	 * Indicates Recent Past View supports the date as a query argument appended to its URL, not as part of a "pretty" URL.
+	 *
+	 * @var bool
+	 */
+	protected static $date_in_url = false;
+
+		// Todo: remove this.
+
+	public function maybe_filter_current_view_slug( $view_slug, $context, $query ) {
+
+		/**
+		 *
+		 *
+		 * @since TBD
+		 *
+		 * @param string $return_value Which value we are going to return as the conversion.
+		 */
+		$prevent_recent_past = apply_filters( 'tribe_events_filter_recent_post_slug', false );
+
+		if ( $prevent_recent_past ) {
+			return $view_slug;
+		}
+
+		if ( tribe_events()->where( 'ends_after', 'now' )->count() === 0 ) {
+
+			return 'recent-past';
+		}
+
+		return $view_slug;
+	}
+
+	/**
+	 * {@inheritDoc}
```

Comment:
:no_entry_sign: Doc comment for parameter "$context" missing

<sub>hash: 7cec6880dfb0481a7b65a42e6d546e6b</sub>